allennlp/training/trainer.py

```diff
@@ -465,39 +465,13 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
 
             # Log parameter values to Tensorboard
             if batch_num_total % self._summary_interval == 0:
-                for name, param in self._model.named_parameters():
-                    self._tensorboard.add_train_scalar("parameter_mean/" + name,
-                                                       param.data.mean(),
-                                                       batch_num_total)
-                    self._tensorboard.add_train_scalar("parameter_std/" + name, param.data.std(), batch_num_total)
-                    if param.grad is not None:
-                        if is_sparse(param.grad):
-                            # pylint: disable=protected-access
-                            grad_data = param.grad.data._values()
-                        else:
-                            grad_data = param.grad.data
-                        self._tensorboard.add_train_scalar("gradient_mean/" + name,
-                                                           grad_data.mean(),
-                                                           batch_num_total)
-                        self._tensorboard.add_train_scalar("gradient_std/" + name,
-                                                           grad_data.std(),
-                                                           batch_num_total)
+                self._parameter_and_gradient_statistics_to_tensorboard(batch_num_total)
                 self._tensorboard.add_train_scalar("loss/loss_train", metrics["loss"], batch_num_total)
                 self._metrics_to_tensorboard(batch_num_total,
                                              {"epoch_metrics/" + k: v for k, v in metrics.items()})
-                # norm of gradients
-                if self._batch_grad_norm is not None:
-                    self._tensorboard.add_train_scalar("gradient_norm",
-                                                       self._batch_grad_norm,
```

Comment:
I think there's no reason to have `_batch_grad_norm` be a field on `self` - you can just return it directly from `self._rescale_gradients` and not pollute the class state with unnecessary fields.  And then pass it in your method call above.